### ISSUE TITLE
Kerberos principal attribute not found on LDAP user - even if kerberos authentication is off

### ISSUE THREAD

**Issue by oculos** [keycloak/keycloak#25294]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ldap

### Describe the bug

I have configured LDAP as my user federation provider. I have disabled any form of kerberos authentication. Still, on the logs, keycloak complains that no kerberos attribute was found.
If kerberos isn't used, why should it look for kerberos attributes?

### Version

23.0.1

### Expected behavior

No error should be produced because an attribute that won't be used isn't found

### Actual behavior

Logs show that "Kerberos principal attribute not found on LDAP user":

```
2023-12-05 12:32:01,614 WARN  [org.keycloak.storage.ldap.LDAPStorageProvider] (executor-thread-11) Kerberos principal attribute not found on LDAP user [uid=francis,cn=users,dc=mydomain,dc=com]. Configured kerberos principal attribute name is [krb5PrincipalName]
```

### How to Reproduce?

Configure LDAP provider
Disable kerberos authentication
Search for a user

**Comment by flixman** [comment]
I am finding the same behavior with 23.0.3, with ldap initialized with the following json file through the API:
 ```json
{
  "name": "ldap",
  "providerId": "ldap",
  "providerType": "org.keycloak.storage.UserStorageProvider",
  "config": {
    "fullSyncPeriod": [
      "-1"
    ],
    "pagination": [
      "true"
    ],
    "usersDn": [
      "$LDAP_BASE_DN"
    ],
    "connectionPooling": [
      "true"
    ],
    "cachePolicy": [
      "DEFAULT"
    ],
    "useKerberosForPasswordAuthentication": [
      "false"
    ],
    "importEnabled": [
      "false"
    ],
    "editMode": [
      "READ_ONLY"
    ],
    "enabled": [
      "true"
    ],
    "changedSyncPeriod": [
      "-1"
    ],
    "bindCredential": [
      "$LDAP_BIND_USER_PASSWORD"
    ],
    "bindDn": [
      "$LDAP_BIND_USER_DN"
    ],
    "usernameLDAPAttribute": [
      "uid"
    ],
    "lastSync": [
      "1575220979"
    ],
    "vendor": [
      "other"
    ],
    "uuidLDAPAttribute": [
      "entryUUID"
    ],
    "connectionUrl": [
      "ldap://$LDAP_HOST:$LDAP_PORT"
    ],
    "allowKerberosAuthentication": [
      "false"
    ],
    "syncRegistrations": [
      "false"
    ],
    "authType": [
      "simple"
    ],
    "customUserSearchFilter": [
      "$LDAP_USER_OBJECT_FILTER"
    ],
    "debug": [
      "false"
    ],
    "searchScope": [
      "$LDAP_SEARCH_SCOPE"
    ],
    "useTruststoreSpi": [
      "ldapsOnly"
    ],
    "priority": [
      "0"
    ],
    "userObjectClasses": [
      "inetOrgPerson, organizationalPerson"
    ],
    "rdnLDAPAttribute": [
      "uid"
    ],
    "validatePasswordPolicy": [
      "false"
    ],
    "batchSizeForSync": [
      "1000"
    ]
  }
}
```

**Comment by a-haurylau** [comment]
Same error on 23.0.4

**Comment by steadyk** [comment]
We are also facing this with version 23.0.4, e.g. on Kiali logins with OIDC.

**Comment by sguilhen** [comment]
I can confirm the issue - caused by https://github.com/keycloak/keycloak/commit/57e51e9dd425d5a9da0064d6380b31ac497f9537#diff-bb4ffa08fae23dc22710d174b81fadf5028b58a8742d9fba4035a2c916e92982R572-R580

The condition there should check first if kerberos auth is enabled, so the `if` clause should look like 

```
if (kerberosConfig.isAllowKerberosAuthentication() && kerberosConfig.getKerberosPrincipalAttribute() != null)
```

This would prevent this warn message to appear when importing the users from LDAP. I will send a PR with a fix shortly.

**Comment by ahus1** [comment]
@sguilhen - although it is "just" a log, it might appear quite often (on each login)? Please consider if you want to provide a backport for this for KC23.

**Comment by sguilhen** [comment]
Yeah, I'll add the label to backport this to 23.

**Comment by Flekyno** [comment]
Hello guys,

please can you confirm that these issue is exactly what i mentioned here: https://github.com/keycloak/keycloak/discussions/26960 ? I'm using keycloak v23.0.3

Seems to me, that the keycloak does not use Kerberos principal attribute: userPrincipalName and instead of try to create kerberos principal name as an combination of samAccountName@realm name which cause problem in situation where upn is different than samAccountName. 

Thanks for answer.
Tomas

**Comment by danfromtitan** [comment]
Although though this was marked as fixed in ver 23.0.7, it's still there:

2024-02-23 00:22:07,791 WARN  [org.keycloak.storage.ldap.LDAPStorageProvider] (executor-thread-98) Kerberos principal attribute not found on LDAP user [cn=user,ou=people,dc=company,dc=domain]. Configured kerberos principal attribute name is [krb5PrincipalName]
